spec/models/service_plan_spec.rb

```diff
@@ -52,7 +52,29 @@
       end
 
       it "shows the modified column is unchanged" do
-        expect(service_plan.modified["schema"]).to eq JSON.parse(valid_ddf)["schema"]
+        expect(service_plan.modified["schema"]).to eq valid_ddf["schema"]
+      end
+    end
+
+    context "modified schema comparison" do
+      let(:service_plan) { create(:service_plan, :base => reordered_ddf) }
+      let(:reordered_ddf) do
+        valid_ddf.dup.tap do |ddf|
+          # pushing the first field onto the end of the array so they're in a different order.
+          field = ddf["schema"]["fields"].shift
+          ddf["schema"]["fields"] << field
+        end
+      end
+
+      it "passes validation when only the order has changed" do
+        service_plan.update!(:modified => reordered_ddf)
+
+        expect(service_plan.valid?).to be_truthy
+      end
+
+      it "fails validation when changing the contents even when the order is different" do
```

Comment:
This reads like another context, and then the `changed_ddf` variable should be promoted to a `let` instead.

Reply:
Sure, that sounds good. It reads better that way too. Updated.